### PR TITLE
Merge HDFS and HDFSIterator Module

### PR DESCRIPTION
**Pull request by krishnadey30**
This PR fixes the following issue https://github.com/chapel-lang/chapel/issues/11067
The following changes have been done:
- Moving `iter HDFSiter` to `HDFS module`
- Removing `HDFSIterator` from makefile
- Replacing `Use HDFSIterator` to `Use HDFS`(whereever needed)

### CODE REVIEW

File: test/studies/HDFS/tzakian/HDFS.chpl [hunk 1]
```diff
@@ -54,3 +58,280 @@ extern proc hdfsOpenFile(hdfsFS: c_void_ptr, path: c_string, flags: c_int, buffe
 /* Returns tSize (int32) */
 extern proc hdfsWrite(hdfsFS: c_void_ptr, hdfsFile: c_void_ptr, buffer: c_string, length: int(32)): int(32);
 
+
+/* ====================== TODO's ========================
```
Comment (ben-albrecht) [inline test/studies/HDFS/tzakian/HDFS.chpl]: Where is this content coming from, and should it be removed elsewhere?

File: modules/packages/HDFSiterator.chpl [hunk 2]
```diff
@@ -1,141 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (krishnadey30) [reply modules/packages/HDFSiterator.chpl]: This content is from `test/studies/HDFS/tzakian/HDFSiterator.chpl`.  I will remove this file.

File: test/studies/HDFS/tzakian/HDFS.chpl (hunk 1 above)
Comment (krishnadey30) [reply test/studies/HDFS/tzakian/HDFS.chpl]: This content is from `test/studies/HDFS/tzakian/HDFSiterator.chpl`.  I will remove this file.

**Comment by krishnadey30** [comment]
> Thanks for your contribution @krishnadey30.
> 
> Additionally, we'll need to add a deprecation test to `tests/deprecation` for the HDFSIterator module.
> 
> Were you able to confirm that the HDFS tests pass with these changes?

Do I need to have HDFS in my system?

**Comment by krishnadey30** [comment]
@ben-albrecht I am getting the following error when I run `chpl test/io/tzakian/hdfs/parBeer/parBeer.chpl`

```
$CHPL_HOME/modules/packages/HDFS.chpl:198: In function 'hdfsChapelConnect':
$CHPL_HOME/modules/packages/HDFS.chpl:205: warning: 'out error: syserr' pattern has been deprecated, use 'throws' function instead
$CHPL_HOME/modules/packages/HDFS.chpl:229: In function 'hdfsOpen':
$CHPL_HOME/modules/packages/HDFS.chpl:236: warning: 'out error: syserr' pattern has been deprecated, use 'throws' function instead
$CHPL_HOME/modules/packages/HDFS.chpl:214: In function 'hdfsChapelDisconnect':
$CHPL_HOME/modules/packages/HDFS.chpl:219: error: const actual is passed to 'ref' formal 'replicatedVar' of rcLocal()
note: to formal of type [ReplicatedDom(1,int(64),false)] c_void_ptr
$CHPL_HOME/modules/packages/HDFS.chpl:229: In function 'hdfsOpen':
$CHPL_HOME/modules/packages/HDFS.chpl:236: error: const actual is passed to 'ref' formal 'replicatedVar' of rcLocal()
note: to formal of type [ReplicatedDom(1,int(64),false)] c_void_ptr
$CHPL_HOME/modules/packages/HDFS.chpl:250: In function 'hdfsClose':
$CHPL_HOME/modules/packages/HDFS.chpl:255: error: const actual is passed to 'ref' formal 'this' of hdfsClose()
note: to formal of type hdfsChapelFile
$CHPL_HOME/modules/packages/HDFS.chpl:252: note: The shadow variable 'this' is constant due to forall intents in this loop
$CHPL_HOME/modules/packages/HDFSiterator.chpl:66: In iterator 'HDFSiter':
$CHPL_HOME/modules/packages/HDFSiterator.chpl:104: error: const actual is passed to 'ref' formal 'this' of hdfsReader()
note: to formal of type hdfsChapelFile
$CHPL_HOME/modules/packages/HDFSiterator.chpl:28: note: The shadow variable 'globalFile' is constant due to task intents in this parallel statement
```

**Comment by ben-albrecht** [comment]
Hey @krishnadey30 - I recently learned that https://github.com/chapel-lang/chapel/issues/11067 is actually no longer needed. Sorry that you got started on it already.

I think we should close this PR now.

**Comment by krishnadey30** [comment]
> Hey @krishnadey30 - I recently learned that #11067 is actually no longer needed. Sorry that you got started on it already.
> 
> I think we should close this PR now.

Ok @ben-albrecht Sir

**Pull request closed without merging**